src/main/java/mate/academy/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -36,4 +36,19 @@ public BookDto getById(Long id) {
                 .orElseThrow(
                     () -> new EntityNotFoundException("Can't find book with such id: " + id)));
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book bookModel = bookMapper.toModel(requestDto);
+        bookModel.setId(id);
+        return bookMapper.toDto(bookRepository.save(bookModel));
+    }
+
+    @Override
+    public void delete(Long id) {
+        if (bookRepository.existsById(id)) {
+            bookRepository.deleteById(id);
+        }
+        throw new EntityNotFoundException("Can't delete book with such id: " + id);

```

Comment:
let's flip this if to follow up with the fail/return early pattern
```suggestion
        if (!bookRepository.existsById(id)) {
            throw new EntityNotFoundException("Can't delete book with such id: " + id);
        }
        bookRepository.deleteById(id);
```